src/utilities/extendSchema.js

```diff
@@ -258,10 +264,103 @@ export function extendSchema(
     if (isUnionType(type)) {
       return extendUnionType(type);
     }
+    if (isEnumType(type)) {
+      return extendEnumType(type);
+    }
+    if (isInputObjectType(type)) {
+      return extendInputObjectType(type);
+    }
     // This type is not yet extendable.
     return type;
   }
 
+  function extendInputObjectType(
+    type: GraphQLInputObjectType,
+  ): GraphQLInputObjectType {
+    return new GraphQLInputObjectType({
+      name: type.name,
+      description: type.description,
+      fields: () => extendInputFieldMap(type),
+      astNode: type.astNode,
```

Comment:
You need to add `extensionASTNodes` here. But first you need to add it into `GraphQLInputObjectType` class.